src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -784,8 +792,8 @@ private void CreateInputOutputPorts()
 
         private void SetInputPorts()
         {
-            //Clear out all the input port models
-            InPorts.Clear();
+
+            InPorts.RemoveAll((p) => { return true; });
```

Comment:
this extension method is used instead because observableCollection has very odd behavior when cleared - there is no way to reference the cleared items and so they cannot be cleaned up properly - this is why the dangling connector was left around in the first place.

Reply:
can you add this comment in the code?

Reply:
sure, will do